fix: chrony PPS implementation, take 2

doudou opened the pull request:
I merged #16 by mistake. Please have a look at it as well.

doudou committed:
fix(cli): mention the PPS period in the documentation of chrony-configure

doudou committed:
fix(cli): display the pulse time instead of the TimeUTC timestamp in `chrony`

doudou committed:
chore: make PPSGuard non-copyable

kapeps commented on src/Main.cpp:
```diff
@@ -81,9 +81,10 @@ int usage()
+        << "  chrony-configure PORT [PERIOD_US]\n"
         << "     permanently configure the device to output the UBLOX time message\n"
-        << "     on PORT that is needed for the chrony subcommand\n"
+        << "     on PORT that is needed for the chrony subcommand. Optionally modify\n"
+        << "     the period of the PPS (in microseconds)\n"
```
You could include that the standard period is 1second

doudou replied:
The intention was to not modify the period at all. Fixed both documentation and code.

doudou committed:
fix(cli): do not modify PPS period if no period is given to chrony-configure

doudou committed:
fix: missing scale factor for PPS seconds to microseconds

doudou commented:
Working version (finally). Trying to get a working mender image now.

doudou committed:
fix: add sanity check on the reception of the TimingPulseData message

doudou committed:
fix: change how the timing information is read in `chrony`

The current strategy was to read a timing information message, and then
wait for the corresponding PPS. The bug here is that we start reading
just after the last PPS, meaning we might be reading the next pulse
information that was received *before* the PPS (because hardware
buffering)

The last commit attempted to solve this particular issue. Which did not
solve the more general issue that if the pps/ublox synchronization
were to have a one or more period errors, we would not be able to
know and detect it.

This commit changes the strategy to poll the timing message, and use
the last timing message received whenever the PPS arrives. This is
not perfect either, but I believe chrony would "solve" outliers in
this case.

doudou committed:
fix: in chrony, handle timeout on pulse information and errors in ublox poll thread

The former is necessary since we won't receive such message until the unit
actually is generating a PPS (so, timeout is actually normal). The latter is
necessary to get the actual error. Throwing in this thread more often
than not kills the process with no message at all.

doudou committed:
fix: improve diagnostic reporting in `chrony`

I was using `\n` instead of endl, which does not flush. Moreover,
test for sequence number and zero time in PPS *before* comparing
to the Ublox information, which makes more sense.

doudou committed:
fix: set PPS_CANWAIT to have time_pps_fetch wait for the PPS

The current implementation is basically a busy-wait, taking
100% CPU

doudou committed:
fix: do not use the same Ublox time message more than once
